phpBB/includes/acp/acp_styles.php

```diff
@@ -259,6 +259,19 @@ protected function action_uninstall()
 		// Get list of styles to uninstall
 		$ids = $this->request_vars('id', 0, true);
 
+		// Don't remove prosilver, you can still deactivate it.
+		$sql = 'SELECT style_id
+			FROM ' . STYLES_TABLE . "
+			WHERE style_name = '" . $this->db->sql_escape('prosilver') . "'";
+		$result = $this->db->sql_query($sql);
+		$prosilver_id = (int) $this->db->sql_fetchrow($result);
+		$this->db->sql_freeresult($result);
+
+		if (in_array($prosilver_id, $ids))
+		{
+			trigger_error($this->user->lang['UNINSTALL_PROSILVER'] . adm_back_link($this->u_action), E_USER_WARNING);
```

Comment:
it would be probably better to use  ```language->lang()``` method instead of ```user->lang``` array for the sake of 3.3/4.0 compatibility.

Reply:
Yes, but the language object is not in use there, still.

I believe this bit should be adressed on another PR if not already exists, like "moving *CPs to controller".

Reply:
Okay, than probably ```user->lang()``` method instead of ```user->lang[]``` array?

Reply:
It's the same as used everywhere there.

Reply:
see the function below

Reply:
@3D-I I see, but the common sense is using new code in new PRs,. But whatever you like.

Reply:
Right, let me work on this one a bit more, thx m8.